src/PlaneWaveBasis.jl

```diff
@@ -121,10 +121,29 @@ build_kpoints(basis::PlaneWaveBasis, kcoords) =
 # This is the "internal" constructor; the higher-level one below should be preferred
 @timing function PlaneWaveBasis(model::Model{T}, Ecut::Number,
                                 kcoords::AbstractVector, ksymops, symops=nothing;
-                                fft_size=determine_grid_size(model, Ecut), variational=true) where {T <: Real}
-    @assert Ecut > 0
+                                fft_size=nothing, variational=true, optimize_fft_size=false, supersampling=2) where {T <: Real}
+    if variational
+        @assert Ecut > 0
+        if fft_size === nothing
+            fft_size = determine_fft_size(model, Ecut; supersampling=supersampling)
+        end
+    else
+        # ensure fft_size is provided, and other options are not set
+        @assert fft_size !== nothing
+        @assert supersampling == 2
+        @assert !optimize_fft_size
+    end
     fft_size = Tuple{Int, Int, Int}(fft_size)
 
+    kpoints = build_kpoints(model, fft_size, kcoords, Ecut; variational=variational)
+
+    if variational && optimize_fft_size
+        fft_size = determine_fft_size_precise(model.lattice, Ecut, kpoints; supersampling=supersampling)
```

Comment:
I'd rather have a common interface to both these methods and use a flag `fast` or `algorithm=:fast or :precise` to switch between both.

Reply:
I had that, but they have different input arguments. This is not very finalized so I'd rather not solidify the API